interfaces/builtin/network_manager.go

```diff
@@ -67,6 +67,29 @@ network packet,
 
 /dev/rfkill rw,
 
+# Needed for modem connections using PPP
+/usr/sbin/pppd ix,
+/etc/ppp/** rwix,
+/dev/ppp rw,
+/dev/tty[^0-9]* rw,
+/run/udev/data/* r,
+/proc/sys/kernel/modprobe r,
+/bin/kmod ix,
+capability sys_module,
```

Comment:
This gives network-manager ring 0 access to the system. If you can load a kernel module, then you can disable all security, etc, etc. We should try to avoid this at all costs and note also that module autoloading should work fine and not require this access. 'snappy config ubuntu-core' in 15.04 allowed you to load kernel modules and IIRC this was exposed in the gadget snap. 16 removed all of this but we should be designing a system where snaps don't load modules themselves. There are a number of ways to handle this that could all be implemented:
- AIUI, 'snap config' will be coming back and it could support module loading
- the gadget snap could support module loading
- the interface could encode a list of modules to load and upon snap install/connect, snapd could load the modules. I discussed this relative to firewall-control here: https://bugs.launchpad.net/snappy/+bug/1583514
- snapd and the snap implementing a particular slot could talk over a carefully designed protocol for having snapd load requested modules (perhaps this is all abstracted behind a rewritten modprobe)

The main points of all of the above is that snaps don't load modules themselves and snapd is expanded to load modules on behalf of snaps, that should be carefully controlled.

@tyhicks - fyi, I think you'll want to keep an eye on this